### PR TITLE
Added eepromdirectory and fixed it to work with new drivers

### PR DESCRIPTION
**Pull request by Sabramz**

### CODE REVIEW

File: Core/Src/eepromdirectory.c [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+        return -1;
+    }
+    /* write data to eeprom given page, offset, and size of data */
+    eeprom_write(address, data, size);
+}
+
+void logFault(uint32_t *fault_code)
+{
+    //Get starting address of faults, then add the number of entries currently in eeprom_faults to find next available address.
+    uint16_t address = eepromGetAddress(const_cast<char*>("FAULTS")) + (sizeof(eeprom_faults) / 4);
```
Comment (Sabramz) [inline Core/Src/eepromdirectory.c:112]: The code here is wrong. I need to find the next available index in eeprom_faults, not the size.

File: Core/Src/eepromdirectory.c [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+        {
+            return eeprom_data[i].id;
+        }
+
+        i++;
+    }
+    return NULL;
+}
+
+void eepromReadData(char *key, void *data, uint16_t size)
```
Comment (Sabramz) [inline Core/Src/eepromdirectory.c:69]: Make all of these return booleans.

File: Core/Src/eepromdirectory.c [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    /* write data to eeprom given page, offset, and size of data */
+    eeprom_write(address, data, size);
+}
+
+void logFault(uint32_t *fault_code)
+{
+    //Get starting address of faults, then add the number of entries currently in eeprom_faults to find next available address.
+    uint16_t address = eepromGetAddress(const_cast<char*>("FAULTS")) + (sizeof(eeprom_faults) / 4);
+
+    /* write the fault code*/
```
Comment (dyldonahue) [inline Core/Src/eepromdirectory.c:114]: in eeprom init: set the fault index to the last byte (whatever it needs to be to get the "else" to trigger

**Review by dyldonahue — CHANGES_REQUESTED** [review]

File: Core/Src/eepromdirectory.c [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+        /* first byte of partition is the index of the most recent fault, faults begin at second byte */
+        *address = startIndex + 1;
+    }
+    else
+    {
+        *address += 4;
+    }
+
+    /* write the fault code*/
+    eepromWriteData(*address, &fault, 4);
```
Comment (Sabramz) [inline Core/Src/eepromdirectory.c:137]: A fault is logged but the first byte in the partition is never updated to the next index. Gotta fix that.

Comment (dyldonahue) [reply Core/Src/eepromdirectory.c:137]: eepromWriteData is not going to/should not be looking for a pointer, just a value.

File: Core/Src/eepromdirectory.c [hunk 5]
```diff
@@ -0,0 +1,171 @@
+#include "eepromdirectory.h"
+#include "general/include/m24c32.h"
```
Comment (Sabramz) [inline Core/Src/eepromdirectory.c:2]: Problem: This file doesn't exist in shepherd.

**Review by dyldonahue — APPROVED** [review]

**Pull request merged**